### PR TITLE
Replace globset with glob-match in watcher.rs and command.rs

### PR DESCRIPTION
**Pull request by abdiu34567**
#### Overview:
This pull request proposes replacing the `globset` library with the `glob-match` library in the `whiz` repository.

#### Changes:
1. **Files Modified**:
   - `watcher.rs`: Implemented `glob-match` logic, replacing all instances of `globset`. Ensured that the watcher functionality continues to operate as expected by testing for file changes.
   - `command.rs`: Removed unused imports related to `globset` and ensured that the integration with `glob-match` is seamless. Also made necessary adjustments to data structures (e.g., replacing `GlobSet` with `Vec<String>` for pattern storage).

2. **Package Changes**:
   - **Added**: `glob-match` – An efficient and fast glob matching library.
   - **Removed**: `globset` – The previous glob matching library.

#### Reason for Change:
The `glob-match` library provides an extremely efficient matching mechanism with zero allocations, no regex compilation, and a linear-time matching process. It's a lightweight and robust alternative to `globset`.

#### Testing:
- All existing unit tests have passed without issues.
- Manual testing was conducted, especially in terms of file watching, and the results were satisfactory.
  
Would appreciate a review to ensure the changes are aligned with the repository's standards and objectives. Feedback is welcome.

### CODE REVIEW

File: src/actors/watcher.rs [hunk 1]
```diff
@@ -78,8 +79,8 @@ impl Actor for WatcherActor {
 #[rtype(result = "()")]
 pub struct WatchGlob {
     pub command: Addr<CommandActor>,
-    pub on: GlobSet,
-    pub off: GlobSet,
+    pub on: Vec<String>,  // Storing raw glob patterns instead of GlobSet
```
Comment (zifeo) [inline src/actors/watcher.rs:82]: Will the comment still be useful for developer once the globset is removed? Same applies to a few comments in the PR.

**Review by zifeo — CHANGES_REQUESTED** [review]
Nice submission. As the change were rather easy, do you see a way to test this end to end?

File: src/actors/watcher.rs (hunk 1 above)
Comment (abdiu34567) [reply src/actors/watcher.rs:82]: Certainly, an end-to-end test would provide greater confidence that the changes work seamlessly from a user's perspective. Here's a potential approach:

Setup: Prepare a dummy directory structure that mimics a typical use case for the application. This would include various files and directories with patterns we intend to test against.

Execution: Run our application on this directory structure. Use different glob patterns to match files and directories, making sure to test edge cases and any patterns that the old globset handled differently than glob-match.

Validation: Verify that the application's output (or behavior) is as expected. This means the right files are detected, the ignored patterns are truly ignored, etc.

Teardown: Clean up any resources or files created during the test.

I can go ahead and set up such a test. Would this approach meet your expectations? If there's a specific testing tool or framework you'd prefer to be used, please let me know!

Comment (zifeo) [reply src/actors/watcher.rs:82]: 👍 that would work !
